src/util/Options.js

```diff
@@ -0,0 +1,169 @@
+ * [guide](https://discordjs.guide/popular-topics/partials.html) for some
+ * important usage information, as partials require you to put checks in place when handling data.
+ * @property {number} [restWsBridgeTimeout=5000] Maximum time permitted between REST responses and their
+ * corresponding websocket events
+ * @property {number} [restTimeOffset=500] Extra time in milliseconds to wait before continuing to make REST
+ * requests (higher values will reduce rate-limiting errors on bad connections)
+ * @property {number} [restRequestTimeout=15000] Time to wait before cancelling a REST request, in milliseconds
+ * @property {number} [restSweepInterval=60] How frequently to delete inactive request buckets, in seconds
+ * (or 0 for never)
+ * @property {number} [restGlobalRateLimit=0] How many requests to allow sending per second (0 for unlimited, 50 for
+ * the standard global limit used by Discord)
+ * @property {string[]|RateLimitQueueFilter} [rejectOnRateLimit] Decides how rate limits and pre-emptive throttles
+ * should be handled. If this option is an array containing the prefix of the request route (e.g. /channels to match any
+ * route starting with /channels, such as /channels/222197033908436994/messages) or a function returning true, a
+ * {@link RateLimitError} will be thrown. Otherwise the request will be queued for later
+ * @property {number} [retryLimit=1] How many times to retry on 5XX errors (Infinity for indefinite amount of retries)
+ * @property {PresenceData} [presence={}] Presence data to use upon login
+ * @property {IntentsResolvable} intents Intents to enable for this connection
+ * @property {WebsocketOptions} [ws] Options for the WebSocket
+ * @property {HTTPOptions} [http] HTTP options
+ */
+
+/**
+ * WebSocket options (these are left as snake_case to match the API)
+ * @typedef {Object} WebsocketOptions
+ * @property {number} [large_threshold=50] Number of members in a guild after which offline users will no longer be
+ * sent in the initial guild member list, must be between 50 and 250
+ */
+
+/**
+ * HTTP options
+ * @typedef {Object} HTTPOptions
+ * @property {number} [version=9] API version to use
+ * @property {string} [api='https://discord.com/api'] Base url of the API
+ * @property {string} [cdn='https://cdn.discordapp.com'] Base url of the CDN
+ * @property {string} [invite='https://discord.gg'] Base url of invites
+ * @property {string} [template='https://discord.new'] Base url of templates
+ * @property {Object} [headers] Additional headers to send for all API requests
+ */
+
+/**
+ * Contains various utilities for client options.
+ */
+class Options extends null {
+  /**
+   * The default client options.
+   * @returns {ClientOptions}
+   */
+  static createDefault() {
+    return {
+      shardCount: 1,
+      makeCache: this.cacheWithLimits({ MessageManager: 200 }),
+      messageCacheLifetime: 0,
+      messageSweepInterval: 0,
```

Comment:
Not an issue with this PR, but in the future we should probably look into implementing a sweeper on all cache types?

Reply:
Yep, we had some talks about more collection types and ways to compose them ergonomically. Future PR.